tools/run_tests/xds_k8s_test_driver/bin/cleanup/cleanup.py

```diff
@@ -337,8 +337,9 @@ def main(argv):
     compute = gcp.compute.ComputeV1(gcp.api.GcpApiManager(), project)
     leakedHealthChecks = []
     for item in compute.list_health_check()['items']:
-        if datetime.datetime.fromisoformat(
-                item['creationTimestamp']) <= get_expire_timestamp():
+        if datetime.datetime.strptime(
+                item['creationTimestamp'],
+                "%Y-%m-%dT%H:%M:%S.%f%z") <= get_expire_timestamp():
```

Comment:
TIL neither `fromisoformat` nor strptime are good enough ISO time parsers.

fromisoformat: https://docs.python.org/3/library/datetime.html#datetime.datetime.fromisoformat

> Caution This does not support parsing arbitrary ISO 8601 strings - it is only intended as the inverse operation of [datetime.isoformat()](https://docs.python.org/3/library/datetime.html#datetime.datetime.isoformat). A more full-featured ISO 8601 parser, [dateutil](https://dateutil.readthedocs.io/en/stable/parser.html#dateutil.parser.isoparse.parser.isoparse) is available in the third-party package dateutil.

And it doesn't support Zulu time:

```
>>> datetime.fromisoformat("2008-09-03T20:56:35.450686Z")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: Invalid isoformat string: '2008-09-03T20:56:35.450686Z'
```

And it python3.6 strptime does is pretty terrible with timezones: https://stackoverflow.com/a/30696682/11697987

```
# zone with `:` separators not supported
>>> datetime.strptime("2008-09-03T20:56:35.450686+04:00", "%Y-%m-%dT%H:%M:%S.%f%z")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/_strptime.py", line 565, in _strptime_datetime
    tt, fraction = _strptime(data_string, format)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/_strptime.py", line 362, in _strptime
    (data_string, format))
ValueError: time data '2008-09-03T20:56:35.450686+04:00' does not match format '%Y-%m-%dT%H:%M:%S.%f%z'

# Zulu not supported
>>> datetime.strptime("2008-09-03T20:56:35.450686Z", "%Y-%m-%dT%H:%M:%S.%f%z")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/_strptime.py", line 565, in _strptime_datetime
    tt, fraction = _strptime(data_string, format)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/_strptime.py", line 362, in _strptime
    (data_string, format))
ValueError: time data '2008-09-03T20:56:35.450686Z' does not match format '%Y-%m-%dT%H:%M:%S.%f%z'
```

Let's use [dateutil.parser.isoparse](https://dateutil.readthedocs.io/en/stable/parser.html#dateutil.parser.isoparse), as recommended in [python docs](https://docs.python.org/3/library/datetime.html#datetime.datetime.fromisoformat).

Reply:
Updated. Using `parse`, not `isoparse`. Let me know if `isoparse` is better.